.github/workflows/myfirst.yml

```diff
@@ -1,12 +1,46 @@
-name: Continuous Integration
-on: [push]
-
+name: CI
+on: 
+  push:
+    paths-ignore:
+    - '**/README.md'
 jobs:
-  build:
-    name: Build and test
+  build-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: docker build --target test --tag my-test-image .
+      - run: docker run my-test-image
+
+  prod-check:
+    #if: ${{ github.ref == 'refs/heads/main' }}
+    if: ${{ github.ref == 'refs/heads/exercise-9' }}
     runs-on: ubuntu-latest
+    needs: build-test
     steps:
-    - uses: actions/checkout@v2
-    #- run: echo Hello World
-    - run: docker build --target test --tag my-test-image .
-    - run: docker run my-test-image
\ No newline at end of file
+      - run: echo "Deploying to production server on branch $GITHUB_REF"
+
+  build-push:
+    runs-on: ubuntu-latest
+    needs: prod-check
+    
+    steps:
+      - uses: actions/checkout@v3
+      #- uses: actions/checkout@5f6796be8905cf35cb55bea0a4e62158c4bf116d 
+      #- uses: actions/checkout@exercise-9
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      - uses: docker/login-action@v3
+        with: 
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - uses: docker/build-push-action@v2
+        with:
+          push: true
+          tags: leonmilcap/m9repo:latest
+          labels: production
```

Comment:
Is `labels` the right argument here? Judging by the [argument list here](https://github.com/docker/build-push-action?tab=readme-ov-file#inputs) I'd say not - we want a `target` attribute to make sure we definitely build our prod image - the label simply adds some metadata (which is no bad thing in itself!)

But I think this does mean that without the build/push below, this stage will actually build your `test` docker stage, since that's the last one defined in the Dockerfile!